frontend/src/features/map_planning/layers/plant/components/PlantLabel.tsx

```diff
@@ -0,0 +1,51 @@
+import { PlantingDto } from '@/bindings/definitions';
+import { ExtendedPlantsSummary } from '@/features/map_planning/layers/plant/components/ExtendedPlantDisplay';
+import { useFindPlantById } from '@/features/map_planning/layers/plant/hooks/useFindPlantById';
+import Konva from 'konva';
+import { useEffect, useRef, useState } from 'react';
+import { Label, Tag, Text } from 'react-konva';
+
+export interface PlantLabelProps {
+  /** Contains plant name that will be displayed on the label. */
+  planting: PlantingDto;
+}
+
+export const PlantLabel = ({ planting }: PlantLabelProps) => {
+  const labelRef = useRef<Konva.Label>(null);
+  const [labelWidth, setLabelWidth] = useState<number>(0);
+
+  useEffect(() => {
+    if (labelRef.current === null || labelWidth) return;
+
+    setLabelWidth(labelRef.current.width());
+  }, [labelRef, labelWidth]);
```

Comment:
I don't think you need labelWidth here, but maybe I am missing something.

Reply:
But you need the if because of the dependency.
I think both are unnecessary.

Reply:
To explain my reasoning, the effect is only run if `labelRef` or `labelWidth` changes.
`labelRef`  only changes on the initial render, but `labelWidth` changes every time `setLabelWidth` is called.
The `if` prevents the infinite rerender loop.
So if we exclude `labelWidth` from the dependencies and the `if` we are still left with the same functionality.

Reply:
Oh ok i thought the extra if check was necessary to avoid infinite loops due to having a setState call inside a useEffect hook.